File: gradle-plugin/groovy-gradle-plugin/buildSrc/src/base/java/com/jamesfchen/IInsertCode.java
package com.jamesfchen;

import java.io.File;
import java.io.InputStream;

/**
 * Copyright ® $ 2017
 * All right reserved.
 *
 * @author: jamesfchen
 * @since: Nov/28/2021  Sun
 */
public interface IInsertCode {
    void onInsertCodeBegin();
    //     byte[] onInsertCodeInDir(ClassInfo info) { return null }
//     byte[] onInsertCodeInJar(ClassInfo info) { return null }
    byte[] onInsertCode(File mather, InputStream classStream, String canonicalName);
    void onInsertCodeEnd();
}

File: gradle-plugin/groovy-gradle-plugin/base-plugin/src/main/kotlin/com/jamesfchen/kotlin/IInsertCode.kt
package com.jamesfchen.kotlin

import java.io.File
import java.io.InputStream

interface IInsertCode {
    fun onInsertCodeBegin()

    //     byte[] onInsertCodeInDir(ClassInfo info) { return null }
//     byte[] onInsertCodeInJar(ClassInfo info) { return null }
    fun onInsertCode(mather: File, classStream: InputStream, canonicalName: String): ByteArray?
    fun onInsertCodeEnd()
}